### ISSUE TITLE
Do we need to use early stopping to prevent over-fitting?

### ISSUE THREAD

**Issue by ruizhaogit** [batra-mlp-lab/visdial#3]
Hi @abhshkdz ,

In your code [train.lua](https://github.com/batra-mlp-lab/visdial/blob/master/train.lua#L79), it seems the model is saved without first evaluated on the validation dataset. Isn't early stopping necessary here?

Thank you! 
Best,
Rui

**Comment by satwikkottur** [comment]
The models are saved irrespective of performance on validation dataset. The user has to cross-verify with the logs in order to chose the right epoch.

**Comment by ruizhaogit** [comment]
Hi, thank you for answering! What do you mean by " cross-verify with the logs"? you mean choose the model based on the training-log manually, and then evaluate it on the validation set and pick the best one?

**Comment by satwikkottur** [comment]
Alternatively, you could use uncommented https://github.com/batra-mlp-lab/visdial/blob/master/train.lua#L84, and evaluate your model every epoch while training to decide early stopping.
(if you just uncomment L84, it evaluates your model every 5 epochs).

**Comment by ruizhaogit** [comment]
Yeah, it is for answer retrieving. How to select the best model on the generate model? Select the one with the lowest validation loss?

**Comment by satwikkottur** [comment]
Notice that we also evaluate generative models on answer retrieval using log-likelihood. We use mean reciprocal rank as our stopping criterion, both for discriminative and generative models.

**Comment by ruizhaogit** [comment]
Thank you!
